src/lib/storage/storage_manager.rb

```diff
@@ -72,7 +68,7 @@ def create_instance(storage_environment = nil)
           storage_environment ||= ::Storage::Environment.new(true)
           create_logger
           log.info("Creating Storage object")
-          @instance = ::Storage::Storage.new(storage_environment)
+          ::Storage::Storage.new(storage_environment)
```

Comment:
**This is wrong.** When calling create_instance from the outside to initialize libstorage with a custom storage environment (e.g. for setting up fake device trees), @instance is not initialized here, so the next call to instance() will create another Storage object, overwriting the previous one with the special environment.